test/core/tsi/ssl_transport_security_test.cc

```diff
@@ -535,310 +146,754 @@ static std::string GenerateTrustBundle() {
+      ASSERT_EQ(tsi_ssl_server_handshaker_factory_create_handshaker(
+                    ssl_fixture->server_handshaker_factory_,
+                    ssl_fixture->network_bio_buf_size_,
+                    ssl_fixture->ssl_bio_buf_size_,
+                    &ssl_fixture->base_.server_handshaker),
+                TSI_OK);
+    }
+
+    static void CheckAlpn(SslTsiTestFixture* ssl_fixture,
+                          const tsi_peer* peer) {
+      ASSERT_NE(ssl_fixture, nullptr);
+      ASSERT_NE(ssl_fixture->alpn_lib_, nullptr);
+      ssl_alpn_lib* alpn_lib = ssl_fixture->alpn_lib_;
+      const tsi_peer_property* alpn_property =
+          tsi_peer_get_property_by_name(peer, TSI_SSL_ALPN_SELECTED_PROTOCOL);
+      if (alpn_lib->alpn_mode != ALPN_CLIENT_SERVER_OK) {
+        ASSERT_EQ(alpn_property, nullptr);
+      } else {
+        ASSERT_NE(alpn_property, nullptr);
+        const char* expected_match = "baz";
+        ASSERT_EQ(memcmp(alpn_property->value.data, expected_match,
+                         alpn_property->value.length),
+                  0);
+      }
+    }
+
+    static void CheckSecurityLevel(const tsi_peer* peer) {
+      const tsi_peer_property* security_level =
+          tsi_peer_get_property_by_name(peer, TSI_SECURITY_LEVEL_PEER_PROPERTY);
+      ASSERT_NE(security_level, nullptr);
+      const char* expected_match = "TSI_PRIVACY_AND_INTEGRITY";
+      ASSERT_EQ(memcmp(security_level->value.data, expected_match,
+                       security_level->value.length),
+                0);
+    }
+
+    static const tsi_peer_property* CheckBasicAuthenticatedPeerAndGetCommonName(
+        const tsi_peer* peer) {
+      const tsi_peer_property* cert_type_property =
+          tsi_peer_get_property_by_name(peer,
+                                        TSI_CERTIFICATE_TYPE_PEER_PROPERTY);
+      EXPECT_NE(cert_type_property, nullptr);
+      EXPECT_EQ(
+          memcmp(cert_type_property->value.data, TSI_X509_CERTIFICATE_TYPE,
+                 cert_type_property->value.length),
+          0);
+      const tsi_peer_property* property = tsi_peer_get_property_by_name(
+          peer, TSI_X509_SUBJECT_COMMON_NAME_PEER_PROPERTY);
+      EXPECT_NE(property, nullptr);
+      return property;
+    }
+
+    static void CheckSessionReusage(SslTsiTestFixture* ssl_fixture,
+                                    tsi_peer* peer) {
+      const tsi_peer_property* session_reused = tsi_peer_get_property_by_name(
+          peer, TSI_SSL_SESSION_REUSED_PEER_PROPERTY);
+      ASSERT_NE(session_reused, nullptr);
+      if (ssl_fixture->session_reused_) {
+        ASSERT_EQ(strncmp(session_reused->value.data, "true",
+                          session_reused->value.length),
+                  0);
+      } else {
+        ASSERT_EQ(strncmp(session_reused->value.data, "false",
+                          session_reused->value.length),
+                  0);
+      }
+    }
+
+    static void CheckServer0Peer(tsi_peer* peer) {
```

Comment:
Can you add a comment above this and `CheckServer1Peer` that they tie to the corresponding certificate files?
The only reason is because I briefly assumed it had something to do with the OpenSSL methods that are appended with `0` and `1`, but it doesn't.

Reply:
Makes sense. Done. Thanks for the review!